pkg/controller/storagecluster/externalstoragecluster_controller.go

```diff
@@ -0,0 +1,265 @@
+			uErr := r.client.Status().Update(context.TODO(), sc)
+			if uErr != nil {
+				reqLogger.Error(uErr, "Failed to update status")
+			}
+			return reconcile.Result{}, err
+		}
+	}
+
+	// All component operators are in a happy state.
+	if r.conditions == nil {
+		reqLogger.Info("No component operator reported negatively")
+		reason := ocsv1.ReconcileCompleted
+		message := ocsv1.ReconcileCompletedMessage
+		statusutil.SetCompleteCondition(&sc.Status.Conditions, reason, message)
+
+		// If no operator whose conditions we are watching reports an error, then it is safe
+		// to set readiness.
+		r := ready.NewFileReady()
+		err = r.Set()
+		if err != nil {
+			reqLogger.Error(err, "Failed to mark operator ready")
+			return reconcile.Result{}, err
+		}
+	} else {
+		// If any component operator reports negatively we want to write that to
+		// the instance while preserving it's lastTransitionTime.
+		// For example, consider the resource has the Available condition
+		// type with type "False". When reconciling the resource we would
+		// add it to the in-memory representation of OCS's conditions (r.conditions)
+		// and here we are simply writing it back to the server.
+		// One shortcoming is that only one failure of a particular condition can be
+		// captured at one time (ie. if resource1 and resource2 are both reporting !Available,
+		// you will only see resource2q as it updates last).
+		for _, condition := range r.conditions {
+			conditionsv1.SetStatusCondition(&sc.Status.Conditions, condition)
+		}
+		reason := ocsv1.ReconcileCompleted
+		message := ocsv1.ReconcileCompletedMessage
+		conditionsv1.SetStatusCondition(&sc.Status.Conditions, conditionsv1.Condition{
+			Type:    ocsv1.ConditionReconcileComplete,
+			Status:  corev1.ConditionTrue,
+			Reason:  reason,
+			Message: message,
+		})
+
+		// If for any reason we marked ourselves !upgradeable...then unset readiness
+		if conditionsv1.IsStatusConditionFalse(sc.Status.Conditions, conditionsv1.ConditionUpgradeable) {
+			r := ready.NewFileReady()
+			err = r.Unset()
+			if err != nil {
+				reqLogger.Error(err, "Failed to mark operator unready")
+				return reconcile.Result{}, err
+			}
+		}
+	}
+	if phaseErr := r.client.Status().Update(context.TODO(), sc); phaseErr != nil {
+		reqLogger.Error(phaseErr, "Failed to update status")
+		return reconcile.Result{}, phaseErr
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func newExternalCephCluster(sc *ocsv1.StorageCluster, cephImage string) *cephv1.CephCluster {
+	labels := map[string]string{
+		"app": sc.Name,
+	}
+	externalCephCluster := &cephv1.CephCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      generateNameForExternalCephCluster(sc),
+			Namespace: sc.Namespace,
+			Labels:    labels,
+		},
+		Spec: cephv1.ClusterSpec{
+			External: cephv1.ExternalSpec{
+				Enable: true,
+			},
+			DataDirHostPath: "/var/lib/rook",
+			CephVersion: cephv1.CephVersionSpec{
+				Image: cephImage,
```

Comment:
```suggestion
```
We don't need CephVersion for External Ceph Cluster.

Reply:
Hope we don't need the `cephImage` as in the comment suggests,
https://github.com/rook/rook/blob/ce1884388543f4dfa445ae6a66a77cb65441d678/cluster/examples/kubernetes/ceph/cluster-external.yaml#L22